### PR TITLE
feat: Add `externalParameters.source` for v1 specs for BYOB

### PR DESCRIPTION
**Pull request by laurentsimon**
closes https://github.com/slsa-framework/slsa-github-generator/issues/2186

### CODE REVIEW

File: .github/workflows/scripts/schedule.actions/verify-verified-token.sh [hunk 1]
```diff
@@ -40,5 +40,6 @@ e2e_verify_predicate_v1_runDetails_builder_id "$PREDICATE_CONTENT" "https://gith
 e2e_verify_predicate_v1_buildDefinition_externalParameters_workflow "$PREDICATE_CONTENT" "$(e2e_this_file_full_path)" "$GITHUB_REF" "git+https://github.com/$GITHUB_REPOSITORY"
 
 # Verify external parameters source
-e2e_verify_predicate_v1_buildDefinition_externalParameters_source "$PREDICATE_CONTENT" "{\"uri\":\"git+https://github.com/$GITHUB_REPOSITORY@$GITHUB_REF\",\"digest\":{\"sha1\":\"$GITHUB_SHA\"}}"
+digest=$(e2e_get_source_sha1)
+e2e_verify_predicate_v1_buildDefinition_externalParameters_source "$PREDICATE_CONTENT" "{\"uri\":\"git+https://github.com/$GITHUB_REPOSITORY@$GITHUB_REF\",\"digest\":{\"sha1\":\"$digest\"}}"
```
Comment (laurentsimon) [inline .github/workflows/scripts/schedule.actions/verify-verified-token.sh:44]: this check should have been failing. The daily runs were not working due to permission errors https://github.com/slsa-framework/slsa-github-generator/actions/runs/5125842921
I've tryied addrssing this in this PR by changing the default perms to `permissions: {}`.

**Comment by ianlewis** [comment]
We had a discussion about adding an identifier for the source repo to the `annotations` field in `resolvedDependencies`? What do you think about that idea?

**Comment by asraa** [comment]
> We had a discussion about adding an identifier for the source repo to the annotations field in resolvedDependencies?

I think if we do, it'd be great to have a universal annotation that's agreed upon between multiple builders, so that parsing the arbitrary `object` can be shared logic.

Since annotations are maps of strings to objects, maybe it can be `usage: <some-source-type-uri>`?

**Comment by ianlewis** [comment]
> I think if we do, it'd be great to have a universal annotation that's agreed upon between multiple builders, so that parsing the arbitrary `object` can be shared logic.

Yeah, I agree. It should probably get into the https://github.com/slsa-framework/github-actions-buildtypes repo at the very least.

> Since annotations are maps of strings to objects, maybe it can be `usage: <some-source-type-uri>`?

For reference, Kubernetes has a similar type of annotations (though string: string) and they use URIs in the keys with "official" annotations starting with using a `kubernetes.io` domain.
https://kubernetes.io/docs/reference/labels-annotations-taints/

**Comment by laurentsimon** [comment]
> Maybe SLSA needs a few of these well known annotations.

I'm curious what's the point of these annotation in practice. iiuc, the verification is only required to look at `externalParameters.source`, and anything in material is to do recursive verification of dependencies...

**Comment by laurentsimon** [comment]
There is WIP for specs to resolve this, so closing for now

**Pull request closed without merging**